app/models/product_user_creator.rb

```diff
@@ -0,0 +1,19 @@
+class ProductUserCreator
+  def self.create(user:, product:, approver:)
+    product_user = ProductUser.new(
+      product: product,
+      user: user,
+      approved_by: approver.id,
+      approved_at: Time.zone.now,
+    )
+
+    product_user.save && manage_training_request(product_user)
```

Comment:
I think it might be a good idea to wrap this in a transaction. So if the training request deletion fails, so does everything.